### ISSUE TITLE
Improvement: changeset feedback message template, customized per editor

### ISSUE THREAD

**Issue by BushmanK** [openstreetmap/openstreetmap-website#1424]
There is a very useful improvement in #1401, however, as far as I understand it, it is unable to solve a case of Maps.me users (a growing fraction of OSM contributors with a questionable input quality). There are several observations, telling us that Maps.me users very rarely respond to changeset comments. Often, they do not seem to be completely aware of OSM, they don't even know that they are registered users of it.

Examining this use case, it is easy to notice, that they receiving a feedback on their edits made in Maps.me from a completely unknown party - some private person, a member of the different project (OSM). There is nothing that connects Maps.me (or any other particular editor or application) to OSM in that letter. So, it is easy to get confused and to ignore this odd message. 

I understand, that it might sound like Maps.me developers should be taking care of it by implementing a notification system in this app. However, this case could be the first but not the last one. So, having a personalized template could be a solution for these future potential cases.

By "personalized template" I mean a template, that could address a user depending on editor he used for making a particular changeset. That might include a custom greeting, like "Dear _whatever editor_ user, ..."; custom documentation links, addressing some common editor-specific issues; custom warning messages,  related to a specific version of an editor (like if we know that a certain version of JOSM has a bug that does something really bad).

**Comment by zerebubuth** [comment]
I think this is an interesting idea, but perhaps providing editor-specific messages might be quite complex. Would it be fair to say that the central issue with contacting Maps.me users is that:

> Often, they do not seem to be completely aware of OSM, they don't even know that they are registered users of it.

I assume this means that they didn't get (or ignored) the sign-up message from OSM? Perhaps that doesn't get sent for OpenID-style signups?

Would a simpler first step be to insert some header / footer text to the email to say something like this?

> You are receiving this email because you contributed a change to map or POI data on OpenStreetMap, possibly through a map app, and another contributor has a comment about it.

**Comment by tomhughes** [comment]
No the message should always be sent - we sometimes don't require confirmation of the email address but I believe the welcome message is always sent.

I don't think for a minute that we should do any kind of editor specific messages which is partly why I changed the title. If we did this at all (and I'm not at all convinced) then I would imagine it would just allow a user to create a set of templates and then choose one by name when starting a message.

**Comment by BushmanK** [comment]
The first rule of UX design is to avoid assuming that users are smart or well-informed. There could be a very long period between the registration email and the first _changeset comment_ (enough to forget about the OSM completely, especially, if they didn't care). A welcome message could have been ignored (as well as any information, showed by the editor app before/during the registration). 

So, the whole purpose of my idea is to bring the reminder exactly to that moment when it's needed the most. To remove any possible factors of confusion (except the user's total ignorance, but that's impossible to deal with it). It also applies to any universal wording like one proposed by @zerebubuth - user has to carefully read and understand that before he'll be able to connect the dots, which is not the case with people, who often assuming that they have just added a personal POI, edited a local copy of map or submitted a feedback to a professional mapping team (just like in case of commercial navigation products). Mentioning an editor/app explicitly in a changeset comment email (not in a comment itself) is a key here. Everything else would be a half-measure.

I don't understand, what exactly means 

> it would just allow a user to create a set of templates and then choose one by name when starting a message

but if it would require any additional actions from one who wants to comment on a changeset (I'm talking about that, not about the personal messages sent via osm.org website), it would also introduce another point of possible human error. If that's is just about personal messages, introducing a template system doesn't make much sense at all, because currently, anyone can just copy and paste a "template" from a text file (and some people actually do that).

However, I can not provide a scientifically correct evaluation of my idea - it is based on my UX development experience (started back in 1998). So, if that would be too hard to implement - it could be better not to waste energy on half-measures.

**Comment by tomhughes** [comment]
So all you're really saying is that you think the email notifying a user of a comment on a changeset should mention the editor that was used to make the change?

That sounds like a much simpler change, and something that much more likely to happen. I'm afraid all your talk of templates made it sound like you wanted something much more complicated than that.

It's probably best to deal with this once #1401 is merged and we can then look to insert an extra sentence in the notification emails that can be looked up from the translation file based on the editor name.

**Comment by BushmanK** [comment]
Yes, that's exactly what I've been talking about. I've mentioned changeset comments in the first paragraph of my initial message as well as in issue title. I might not have said that clear enough.
